src/main/java/org/openrewrite/staticanalysis/EqualsAvoidsNull.java

```diff
@@ -75,20 +80,40 @@ public TreeVisitor<?, ExecutionContext> getVisitor() {
                         new UsesMethod<>(CONTENT_EQUALS)),
                 new JavaVisitor<ExecutionContext>() {
                     @Override
-                    public J visitMethodInvocation(J.MethodInvocation method, ExecutionContext ctx) {
-                        J.MethodInvocation m = (J.MethodInvocation) super.visitMethodInvocation(method, ctx);
-
-                        if (!isStringComparisonMethod(m) || !hasCompatibleArgument(m) || m.getSelect() instanceof J.Literal) {
-                            return m;
-                        }
+                    public J.Block visitBlock(J.Block block, ExecutionContext ctx) {
```

Comment:
The variable declaration lookup kind of works:

![Variable Declaration Lookup](https://github.com/user-attachments/assets/84faf3c5-4e02-4a5a-9e17-a3a2227208f6)

Now the timing is currently wrong as `visitBlock` gets called after the `visitMethodInvocation` function, which then needs to check the pre-built `VARIABLE_DECLARATIONS` cache to determine which ones to flip and which ones not to. I'm sure you guys can help me or easily take over to add the missing parts.

Reply:
what do you think, is this any good? @greg-at-moderne @timtebeek